src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependencyNode.cs

```diff
@@ -126,6 +127,32 @@ internal DependencyNode()
         public string Caption { get; set; }
 
         /// <summary>
+        /// Actual formal name of the node. In most cases it will be equal to Caption,
+        /// however in some cases caption would have name + version or something else.
+        /// So to allow tree provider store formal name we need this property too. 
+        /// It is a hack now and we should make it public and move to IDependencyNode when
+        /// we get a chance and it is safe to do a breaking change.
```

Comment:
Additionally, when will be a better time to do this that we can make breaking changes? I find it highly unlikely we'll be able to make major breaking changes after we ship 1.0, so we should make this public now if we can.